reviewed some of the config template js code and added full screen mode

abenari opened the pull request:
1. changed the call for on content load from manually calling a method to triggering an event, that change allows pages with additional js (such as config template) to add it's own onContent Load implementation.
2. Added a full screen mode for the template editor.

ohadlevy commented on app/assets/javascripts/config_template.js:
```diff
@@ -1,80 +1,106 @@
-      if($('.diffMode').size() >0){ set_edit_mode( $(".template_text")); }
-    })
-  }
+jQuery.fn.exists = function(){return ($(this).length > 0);}
```
should this be in application instead of the template.js?

ohadlevy commented on app/assets/javascripts/config_template.js:
```diff
@@ -82,6 +108,33 @@ function create_editor(item) {
+  $('.navbar').addClass('hidden');
+  $('.logo-bar').addClass('hidden');
+  $editor.resize();
+  $('#main').append("<div class='exit-fullscreen'><a class='btn btn-large' onclick='exit_fullscreen()' href='#'><i class='icon-resize-small'></a></i></div>");
```
I'm missing hover text

ohadlevy commented:
Looks generally good, please however break it into two commits as its hard to follow the changes and amend the commit message to include the issue number.. thanks!

abenari committed:
changed the on content load direct method call to event
this change allow extending the on content load easly.

abenari committed:
changed the on load anonymus function in the config template js
to safer more simple document level event bindings.

abenari committed:
fixes #3274 moved config template to the new two-pane and add full screen editing.

abenari committed:
fixed comments.
1. moved the exists() method to application.js
2. added on hover tool tips for the new buttons.

ohadlevy commented on app/views/config_templates/_form.html.erb:
```diff
@@ -90,6 +97,7 @@
       </div>
     </div>
     <%= submit_or_cancel f %>
+  <div class='exit-fullscreen' style='display:none;'><a class='btn btn-large' onclick='exit_fullscreen()' href='#' title="<%= _('Exit full screen') %>"><i class='icon-resize-small'></i></a></div>
```
out of interest, why style vs class=hide?

ohadlevy commented on app/views/config_templates/index.html.erb:
```diff
@@ -1,3 +1,6 @@
+<%= javascript 'config_template', 'ace/ace',
```
instead of duplicating? should we move to a helper or something?

abenari committed:
fixed comments

ohadlevy commented:
merged as 1e1b2f1 and 5e2b847 thanks @abenari
